veps/sig-storage/declarative-volume-hotplug.md

```diff
@@ -0,0 +1,352 @@
+- The subresource API is not compatible with a [GitOps](https://kubevirt.io/user-guide/operations/gitops/) workflow
+- The subresource API is difficult for a user to invoke directly, so tools like `virtctl` are required
+
+## Goals
+
+- Address shortcomings of the Virtual Machine subresource API by allowing for Virtual Machine volumes to be updated in a declarative way and have those changes applied immediately
+- Support empty CD-ROM disks (no corresponding volume)
+- Inject/eject of CD-ROM volumes
+- A soft goal is to refactor the subresource API implementation to simply call the new declarative API and deprecate `vm.status.volumeRequests`
+
+## Non Goals
+
+- This feature relies on existing [volume hotplug machinery](https://kubevirt.io/user-guide/storage/hotplug_volumes/), so live updates are limited to DataVolume and PersistentVolumeClaim [volume types](https://kubevirt.io/user-guide/storage/disks_and_volumes/#volumes). Support for other volume types may be added in the future
+
+## Definition of Users
+
+- End Users: these are people/programs that have permission to update Virtual Machine specifications
+
+## User Stories
+
+- As a KubeVirt user, I want to be able to use `kubectl edit` to add/remove/change the volumes of a Virtual Machine and have those changes take effect immedietly without restarting the VM
+- As a KubeVirt user, I want to be able to add/remove/change the volumes of a Virtual Machine by updating the VM definition, pushing it to a git repository, and when a system like [Open Cluster Management](https://open-cluster-management.io/) or [ArgoCD](https://argoproj.github.io/cd/) applies those changes, they will take effect immedietly without restarting the VM
+- As a KubeVirt user, I want to be able define a CD-ROM disk on a Virtual Machine and simulate injecting/ejecting a CD-ROM disk by adding/removing volumes from the VM specification. Those changes should take effect immedietly without starting the Virtual Machine
+
+## Repos
+
+kubevirt/kubevirt
+
+## Design
+
+### Enabling Declarative Volume Hotplug
+
+A couple different options were considered.
+
+#### Option 1 - LiveUpdate VM Rollout Strategy + Hotplug Volume Update Strategy
+
+This option leverages [VM Rollout Strategy](https://github.com/kubevirt/community/blob/main/design-proposals/vm-rollout-strategy/vm-rollout-strategy.md) and [Volume Update Strategy](https://github.com/kubevirt/community/blob/main/design-proposals/volume-update-strategy.md)
+
+- A new Volume Update Strategy called `HotPlug` will be created
+- Declarative Volume update will be enabled when the VM Rolout Strategy is `LiveUpdate` and Volume Update Strategy is `Hotplug`
+- If either of those values are not set, the volume update will not be applied until the Virtual Machine is restarted
+
+Pros:
+- Backward compatible
+- Users can opt in
+
+Cons:
+- Cannot achieve soft goal of having the subresource API leverage the declarative API
+
+#### Option 2 - New default (most of the time)
```

Comment:
So, we don't have cases where the  user might desire that the addition of the volume happen at restart? If this isn't case, then I'm in favor to keep the options as simple as possible and this seems a good choice.

Reply:
I cannot think of a case where the user would declaratively set a volume `hotpluggable: true` and not want to use it right away. But that is how it works now. Maybe it doesn't make sense to maintain this behavior